src/Weather_forecast_on_dashboard.py

```diff
@@ -0,0 +1,170 @@
+import streamlit as st
+import pandas as pd
+import plotly.express as px
+from herbie import FastHerbie
+from datetime import datetime, timedelta
+import numpy as np
+import os
+
+# Define the points with the provided coordinates and create a human-readable location column
+points = pd.DataFrame(
+    {
+        "latitude": [
+            27.035, 27.188, 27.085, 27.055, 27.186, 27.138, 26.97, 26.898,
+            26.806, 26.706, 26.698, 26.708, 26.679, 26.8, 26.704, 26.5,
+            26.566, 26.679, 26.201, 26.501, 26.463, 26.718, 26.63, 24.142,
+            23.956, 23.657
+        ],
+        "longitude": [
+            70.515, 70.661, 70.638, 70.72, 70.81, 71.024, 70.917, 70.99599,
+            70.732, 70.81, 70.875, 70.982, 71.027, 71.128, 71.12699, 71.285,
+            71.369, 71.452, 71.295, 72.512, 72.836, 73.049, 73.581, 74.73099,
+            74.625, 74.772
+        ],
+        "stid": [
+            "aa", "bb", "cc", "dd", "ee", "ff", "gg", "hh", "ii", "jj", "kk", "ll",
+            "mm", "nn", "oo", "pp", "qq", "rr", "ss", "tt", "uu", "vv", "ww", "xx", "yy", "zz"
+        ]
+    }
+)
+
```

Comment:
I think the input lat lon should be on the page

Reply:
So, manually input lat lon rather than having pre-selected sites?
If yes, then there's two ways i can approach it. keep all these sites + add comment box to add lat and lon.
Or completely remove these sites, and let the user select them.

Reply:
updated the code, so users can now choose which location they want the data for.